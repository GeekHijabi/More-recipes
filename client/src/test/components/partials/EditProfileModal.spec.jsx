import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { EditProfileModal }
  from '../../../js/components/Partials/EditProfileModal';

const userInput = {
  userName: 'test',
  bio: 'testing',
  summary: 'tester',
  imageUrl: ''
};

const props = {
  editProfile: jest.fn(() => Promise.resolve({ ok: true })),
  currentUser: {},
  toggle: jest.fn(() => Promise.resolve({ ok: true })),
  isOpen: true,
  isLoadingRecipe: true
};
describe('Component: EditProfileModal', () => {
  beforeEach(() => {
    global.toastr = {
      success: () => {},
      error: () => {}
    };
  });
  it('should recieve next props', () => {
    const spy = sinon.spy(EditProfileModal.prototype, 'componentWillReceiveProps');
    const nextProps = {
      currentUser:
        {
          userName: 'test',
          bio: 'testing',
          summary: 'tester',
          imageUrl: ''
        },

    };
    const wrapper = shallow(<EditProfileModal {...nextProps} />);
    wrapper.setProps({ prop: nextProps });
    expect(spy.calledOnce).toEqual(true);
  });
  describe('EditProfileModal snapshot', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<EditProfileModal {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('onChange', () => {
    it('should change event', () => {
      const event = {
        target: { name: 'summary', value: 'tester' },
      };
      const wrapper = shallow(<EditProfileModal {...props} />);
      wrapper.instance().onChange(event);
      expect(wrapper.instance().state.summary).toBe('tester');
    });
  });

  describe('onSubmit', () => {
    it('should add recipe when recipe details is set to the state', () => {
      const event = {
        preventDefault: jest.fn()
      };
      const submit = shallow(<EditProfileModal {...props} />);
      submit.setState(userInput);
      submit.setState({ hasError: false });
      submit.instance().onSubmit(event);
    });
  });
});
