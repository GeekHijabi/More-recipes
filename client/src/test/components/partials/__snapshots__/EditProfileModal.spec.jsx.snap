// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: EditProfileModal EditProfileModal snapshot should render correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  toggle={[MockFunction]}
  zIndex={1050}
>
  <ModalHeader
    closeAriaLabel="Close"
    tag="h4"
    toggle={[MockFunction]}
    wrapTag="div"
  >
    Edit Profile
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <Form
      tag="form"
    >
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleEmail"
          sm={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Username
        </Label>
        <Col
          sm={8}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            id="exampleEmail"
            name="lastName"
            onChange={[Function]}
            placeholder="input last name"
            tag="p"
            type="text"
            value=""
          />
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleEmail"
          sm={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Bio
        </Label>
        <Col
          sm={8}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            id="exampleEmail"
            name="bio"
            onChange={[Function]}
            placeholder="Input a bio describing you"
            tag="p"
            type="text"
            value=""
          />
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleEmail"
          sm={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Summary
        </Label>
        <Col
          sm={8}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            id="exampleEmail"
            name="summary"
            onChange={[Function]}
            placeholder="Input a cute summary of yourself"
            tag="p"
            type="text"
            value=""
          />
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleFile"
          lg={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Image
        </Label>
        <Col
          sm={8}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <input
            accept="image/*"
            name="image"
            onChange={[Function]}
            type="file"
          />
          <img
            alt="sample"
            className="styledImage"
            src={Object {}}
          />
        </Col>
      </FormGroup>
      <FormGroup
        check={true}
        row={true}
        tag="div"
      >
        <Col
          className="modal-button"
          sm={
            Object {
              "offset": 2,
              "size": 10,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Button
            className="submit-btn"
            color="secondary"
            onClick={[Function]}
            tag="button"
          >
            Edit
          </Button>
        </Col>
      </FormGroup>
    </Form>
  </ModalBody>
</Modal>
`;
